src/pages/SearchPage.js

```diff
@@ -35,207 +32,195 @@ const propTypes = {
 
-    componentDidUpdate(prevProps) {
-        if (_.isEqual(prevProps.reports, this.props.reports) && _.isEqual(prevProps.personalDetails, this.props.personalDetails)) {
+    useEffect(() => {
+        if (!isMounted.current) {
+            isMounted.current = true;
             return;
         }
-        this.updateOptions();
-    }
 
-    onChangeText(searchValue = '') {
-        Report.searchInServer(searchValue);
-        this.setState({searchValue}, this.debouncedUpdateOptions);
-    }
+        debouncedUpdateOptions();
+        // Ignoring the rule intentionally, we want to run the code only when search Value changes to prevent additional runs.
+        // eslint-disable-next-line react-hooks/exhaustive-deps
+    }, [searchValue]);
 
     /**
      * Returns the sections needed for the OptionsSelector
      *
      * @returns {Array}
      */
-    getSections() {
+    const getSections = () => {
         const sections = [];
         let indexOffset = 0;
 
-        if (this.state.recentReports.length > 0) {
+        if (searchOptions.recentReports.length > 0) {
             sections.push({
-                data: this.state.recentReports,
+                data: searchOptions.recentReports,
                 shouldShow: true,
                 indexOffset,
             });
-            indexOffset += this.state.recentReports.length;
+            indexOffset += searchOptions.recentReports.length;
         }
 
-        if (this.state.personalDetails.length > 0) {
+        if (searchOptions.personalDetails.length > 0) {
             sections.push({
-                data: this.state.personalDetails,
+                data: searchOptions.personalDetails,
                 shouldShow: true,
                 indexOffset,
             });
-            indexOffset += this.state.recentReports.length;
+            indexOffset += searchOptions.recentReports.length;
         }
 
-        if (this.state.userToInvite) {
+        if (searchOptions.userToInvite) {
             sections.push({
-                data: [this.state.userToInvite],
+                data: [searchOptions.userToInvite],
                 shouldShow: true,
                 indexOffset,
             });
         }
 
         return sections;
-    }
+    };
 
-    searchRendered() {
+    const searchRendered = () => {
         Timing.end(CONST.TIMING.SEARCH_RENDER);
         Performance.markEnd(CONST.TIMING.SEARCH_RENDER);
-    }
-
-    updateOptions() {
-        const {recentReports, personalDetails, userToInvite} = OptionsListUtils.getSearchOptions(
-            this.props.reports,
-            this.props.personalDetails,
-            this.state.searchValue.trim(),
-            this.props.betas,
-        );
-        this.setState({
-            userToInvite,
-            recentReports,
-            personalDetails,
-        });
-    }
+    };
+
+    const onChangeText = (value = '') => {
+        Report.searchInServer(searchValue);
```

Comment:
This is causing the previously stored `searchValue` to be sent and not the newest `value`